apps/full-site-editing/full-site-editing-plugin/full-site-editing/templates/class-wp-template-inserter.php

```diff
@@ -98,6 +98,31 @@ public function fetch_template_parts() {
 		}
 	}
 
+	/**
+	 * Retries a call to wp_remote_get on error.
+	 *
+	 * @param string $request_url Url of the api call to make.
+	 * @param string $request_args Addtional arguments for the api call.
+	 * @param int    $attempt The number of the attempt being made.
+	 * @return array wp_remote_get reponse array
+	 */
+	private function fetch_retry( $request_url, $request_args = null, $attempt = 1 ) {
+		$max_retries = 3;
+
+		$response = wp_remote_get( $request_url, $request_args );
+
+		if ( ! is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		if ( $attempt > $max_retries ) {
```

Comment:
Just for the sake of clarity, what about `>=`?
It kinda triggers me that `$max_retries = 3` but it performs 4 attempts. 😄

Reply:
the joys of language - I had read @gwwar's comment '_Maybe we can do max three retries on error_' as  '_if the first attempt fails retry it 3 more times_',  so:

`1 attempt + 3 retries = 4 attempts` 😄

Happy to wind this back if people think 3 attempts in total is sufficient.